src/Storage/Models/InstanceQueryParameters.cs

```diff
@@ -0,0 +1,542 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Altinn.Platform.Storage.Helpers;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+
+namespace Altinn.Platform.Storage.Models
+{
+    /// <summary>
+    /// Represents members to filter, sort, and manage instances.
+    /// </summary>
+    public class InstanceQueryParameters
+    {
+        private const string _appIdParameterName = "appId";
+        private const string _appIdsParameterName = "appIds";
+        private const string _archiveReferenceParameterName = "archiveReference";
+        private const string _continuationTokenParameterName = "continuationToken";
+        private const string _creationDateParameterName = "created";
+        private const string _currentTaskParameterName = "process.currentTask";
+        private const string _dueBeforeParameterName = "dueBefore";
+        private const string _excludeConfirmedByParameterName = "excludeConfirmedBy";
+        private const string _instanceOwnerIdentifierHeaderName = "X-Ai-InstanceOwnerIdentifier";
+        private const string _instanceOwnerPartyIdParameterName = "instanceOwner.partyId";
+        private const string _lastChangedParameterName = "lastChanged";
+        private const string _messageBoxIntervalParameterName = "msgBoxInterval";
+        private const string _orgParameterName = "org";
+        private const string _processEndEventParameterName = "process.endEvent";
+        private const string _processEndedParameterName = "process.ended";
+        private const string _processIsCompleteParameterName = "process.isComplete";
+        private const string _searchStringParameterName = "searchString";
+        private const string _sizeParameterName = "size";
+        private const string _sortByParameterName = "sortBy";
+        private const string _statusIsActiveOrSoftDeletedParameterName = "status.isActiveOrSoftDeleted";
+        private const string _statusIsArchivedOrSoftDeletedParameterName = "status.isArchivedOrSoftDeleted";
+        private const string _statusIsArchivedParameterName = "status.isArchived";
+        private const string _statusIsHardDeletedParameterName = "status.isHardDeleted";
+        private const string _statusIsSoftDeletedParameterName = "status.isSoftDeleted";
+        private const string _visibleAfterParameterName = "visibleAfter";
+
+        /// <summary>
+        /// Gets or sets the application identifier.
+        /// </summary>
+        [FromQuery(Name = _appIdParameterName)]
```

Comment:
Why are the properties decorated with FromQuery and FromHeader? There's no data binding involved here?

Reply:
[The BuildQueryString ](https://github.com/Altinn/altinn-storage/blob/d0dba6415b8dcf0a520dde0deb06f8cc8901619d/src/Storage/Models/InstanceQueryParameters.cs#L209) is using them when building the query string. It uses them to avoid using the properties which belong to the type itself in the query string.

We can delete these attributes and the `BuildQueryString` if we kept the original `BuildRequestLink`

Reply:
We are agreed to keep using this binding because it is the best practice.